### ISSUE TITLE
Does the library have to require ciso8601?

### ISSUE THREAD

**Issue by bullettrainio** [influxdata/influxdb-client-python#98]
I'm using this library in my project, but the dependency on ciso8601 is really painful. Now my docker builds need to install GCC and pythen-dev in order to satisfy `pipenv install` as part of my docker builds. This blows put the size of my docker images, and significantly increases build times. 

It also means its harder for people to contribute to my project as it makes their local development environment more complex. 

I don't really care about date-parsing speed in my case. I would suggest having some option that doesn't depend on ciso8601.

**Comment by bednar** [comment]
Hi @bullettrainio,

thanks for use our client.

We would like to keep the `ciso8601` as a default dependency, because it's improves data parsing a lot of.

Are you able to configure your Docker builds to install client with a custom `requirements.txt`?

Something like:

```bash
pip install influxdb-client --no-deps --requirement requirements.txt
```

##### requirements.txt - without ciso8601

```txt
rx >= 3.0.1
certifi >= 14.05.14
six >= 1.10
python_dateutil >= 2.5.3
setuptools >= 21.0.0
urllib3 >= 1.15.1
pytz>=2019.1
```

We could change the date-parsing to be tolerant to missing `ciso8601`.

Is it suitable for you?

Regards

**Comment by dabeeeenster** [comment]
Hmm yes it would, but in this case we're using pipenv which doesnt have the ability to do `--no-deps`. I am looking to do a multi stage build in docker which will negate much of this pain. 

Having a separate library that doesnt have the ciso dependency would be the ideal, here. But I realise that's not very clean for you guys.

**Comment by bednar** [comment]
Yeah, separate library will be too painful for us 😞

**Comment by bednar** [comment]
Hi @bullettrainio, 

The issue is fixed in [1.9.0 milestone](https://github.com/influxdata/influxdb-client-python/milestone/11). 

If you would like to use a dev version without ciso then install client via:

```pip install git+https://github.com/influxdata/influxdb-client-python.git@master```

Regards

**Comment by dabeeeenster** [comment]
Ah thats awesome thanks
